src/PuliRunner.php

```diff
@@ -70,15 +70,24 @@ public function __construct($binDir = null)
     /**
      * Runs a Puli command.
      *
-     * @param string $command The Puli command to run.
+     * @param string   $command The Puli command to run.
+     * @param string[] $args    Arguments to quote and insert into the command.
+     *                          For each key "key" in this array, the placeholder
+     *                          "%key%" should be present in the command string.
      *
      * @return string The lines of the output.
      */
-    public function run($command)
+    public function run($command, array $args = array())
     {
+        $replacements = array();
+
+        foreach ($args as $key => $arg) {
+            $replacements['%'.$key.'%'] = escapeshellarg($arg);
```

Comment:
you should use the util class from the Process component for that. ``escapeshellarg`` does not escape things fully on Windows